### PR TITLE
async display of linting results (#190)

### PR DESCRIPTION
**Pull request by jabooth**
Updates listing results on the fly as each linter reports rather than waiting for all at once (closing #190)

On a large file:
async_lint

### CODE REVIEW

**Comment by DonJayamanne** [comment]
@jabooth , sweet and simple.

File: src/client/providers/lintProvider.ts [hunk 1]
```diff
@@ -114,27 +114,29 @@ export class LintProvider extends vscode.Disposable {
...
+
+        promises.forEach(p => {
+            p.then(msgs => {
+                if (cancelToken.token.isCancellationRequested) {
+                    return;
+                }
+                
+                // Build the message and suffix the message with the name of the linter used
+                msgs.forEach(d => {
+                    d.message = `${d.message}`;
```
Comment (jabooth) [inline src/client/providers/lintProvider.ts]: @DonJayamanne I kept this line from the previous code but it doesn't make sense to me - as `d.message` is a `string`, surely:
```js
d.message === `${d.message}`
```
so this is a no-op?

**Comment by DonJayamanne** [comment]
@jabooth , ha ha... I know how that got in there. I just remove it. Thanks again

**Comment by jabooth** [comment]
@DonJayamanne haha cool, no worries! Just wanted to check I wasn't missing something super subtle 😅

**Pull request merged**